scala/lambdas/postingest-message-resender/src/main/scala/uk/gov/nationalarchives/postingestresender/Lambda.scala

```diff
@@ -0,0 +1,103 @@
+import uk.gov.nationalarchives.DADynamoDBClient.{DADynamoDbRequest, given}
+import uk.gov.nationalarchives.dynamoformatters.DynamoFormatters
+import uk.gov.nationalarchives.dynamoformatters.DynamoFormatters.*
+import uk.gov.nationalarchives.postingestresender.Lambda.{Config, Dependencies, QueueMessage}
+import uk.gov.nationalarchives.utils.EventCodecs.given
+import uk.gov.nationalarchives.utils.PostingestUtils.Queue
+import uk.gov.nationalarchives.utils.{Generators, LambdaRunner}
+import uk.gov.nationalarchives.{DADynamoDBClient, DASQSClient}
+
+import java.time.Instant
+import java.util.UUID
+
+class Lambda extends LambdaRunner[ScheduledEvent, Unit, Config, Dependencies] {
+  given Encoder[QueueMessage] = deriveEncoder[QueueMessage]
+
+  override def dependencies(config: Config): IO[Dependencies] = IO(
+    Dependencies(DADynamoDBClient[IO](), DASQSClient[IO](), () => Generators().generateInstant)
+  )
+
+  override def handler: (ScheduledEvent, Config, Dependencies) => IO[Unit] = { (triggerEvent, config, dependencies) =>
+
+    /** This function receives a queue as a parameter. For the particular queue, it retrieves the details from the post ingest table, it also retrieves the 'Message Retention
+      * Period' for this queue. If any of the items in the post ingest table has a 'lastQueued' time before the message retention period, it resends the message to the queue and
+      * updates the 'lastQueued' value for this item in post ingest table with the current datetime.
+      */
+
+    def resendExpiredMessages(queue: Queue): IO[Unit] = for {
+      queueAttributes <- dependencies.sqsClient.getQueueAttributes(queue.queueUrl, List(QueueAttributeName.MESSAGE_RETENTION_PERIOD))
+      messageRetentionPeriod: Long = queueAttributes.attributes().get(QueueAttributeName.MESSAGE_RETENTION_PERIOD).toLong
+      dateTimeNow: Instant = dependencies.instantGenerator()
+      dateTimeCutOff: Instant = dateTimeNow.minusSeconds(messageRetentionPeriod)
+
+      expiredItems <- dependencies.dynamoClient.queryItems[PostIngestStateTableItem](
+        config.dynamoTableName,
+        AndCondition(DynamoFormatters.queue === queue.queueAlias, lastQueued < dateTimeCutOff.toString),
+        Some("QueueLastQueuedIdx")
+      )
+
+      _ <- if expiredItems.isEmpty then IO.unit else logger.info(s"Resending ${expiredItems.size} items to '${queue.queueAlias}' queue")
+      _ <- expiredItems
+        .parTraverseN(25) { item =>
+          sendToQueue(item, queue) >>
+            updateLastQueued(item, queue, dateTimeNow)
+        }
+        .void
+    } yield ()
+
+    def sendToQueue(item: PostIngestStateTableItem, queue: Queue): IO[Unit] =
+      dependencies.sqsClient
+        .sendMessage(queue.queueUrl)(QueueMessage(item.assetId, item.batchId, queue.resultAttrName, item.input))
+        .handleErrorWith { error =>
+          logger.error(s"""Failed to send message for assetId '${item.assetId}' to '${queue.queueAlias}' queue:
```

Comment:
This log line won't be called because you're not flatMapping it. You need to do 
```scala
logger.error(s"""Failed to send message for assetId '${item.assetId}' to '${queue.queueAlias}' queue:
                      |${error.getMessage}""".stripMargin) >> IO.raiseError(error)
```